File: Python/leetCode242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        check = dict()
        for c in s:
            try:
                check[c] += 1
            except KeyError:
                check[c] = 1

        for c in t:
            try:
                check[c] -= 1
            except KeyError:
                check[c] = -1

        for f in check.values():
            if f!=0:
                return False
        return True

if __name__=="__main__":
    s = "anagram"
    t = "nagaram"
    solution = Solution()
    print(solution.isAnagram(s, t))

File: leetCode242.c
#include<stdio.h>
#include<string.h>

int isAnagram(char * s, char * t)
{
	int lens = strlen(s);
	int lent = strlen(t);
	
	if (lens != lent)
	{
		return 0;
	}
	
	int flag[26];
	memset(flag, 0, sizeof(flag));
	
	int i;
	for (i=0; i<lens; i++)
	{
		flag[s[i]-'a'] ++;
		flag[t[i]-'a'] --;
	}
	for (i=0; i<26; i++)
	{
		if (flag[i] != 0)
		{
			return 0;
		}
	}
	
	return 1;
}

int main(void)
{
	char s[10] = "anagram";
	char t[10] = "nagaram";
	int ans = isAnagram(s, t);
	printf("%d", ans);
} 
